pkg/pushsync/pushsync.go

```diff
@@ -191,7 +192,8 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 
 	// forwarding replication
 	storedChunk := false
-	if ps.topologyDriver.IsWithinDepth(chunk.Address()) {
+	inNeighborhood := ps.topologyDriver.IsWithinDepth(chunk.Address())
```

Comment:
`storedChunk` should be propagated to `pushToClosest` in the case that the node already stored it.